src/platform/Linux/BLEManagerImpl.cpp

```diff
@@ -215,12 +219,29 @@ void BLEManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event)
         break;
 
     case DeviceEventType::kCHIPoBLEWriteReceived:
-        HandleWriteReceived(event->CHIPoBLEWriteReceived.ConId, &CHIP_BLE_SVC_ID, &ChipUUID_CHIPoBLEChar_RX,
-                            PacketBufferHandle::Adopt(event->CHIPoBLEWriteReceived.Data));
+#if !CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION
+        // If BLE has been stopped ignore any write messages as they cannot go anywhere
+        if (GetBleTerminating())
+        {
+            ChipLogProgress(DeviceLayer, "Non-concurrent mode BLE stopped, BLE write write ignored");
+        }
+        else
+#endif
+        {
+            HandleWriteReceived(event->CHIPoBLEWriteReceived.ConId, &CHIP_BLE_SVC_ID, &ChipUUID_CHIPoBLEChar_RX,
+                                PacketBufferHandle::Adopt(event->CHIPoBLEWriteReceived.Data));
+        }
         break;
 
     case DeviceEventType::kCHIPoBLEIndicateConfirm:
         HandleIndicationConfirmation(event->CHIPoBLEIndicateConfirm.ConId, &CHIP_BLE_SVC_ID, &ChipUUID_CHIPoBLEChar_TX);
+#if !CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION
+        if (GetBleTerminating())
+        {
+            ChipLogProgress(DeviceLayer, "Non-concurrent mode BLE packet acknowledged. Sending CloseAllBLEConnections");
```

Comment:
Isn't this assuming that one indication will be enough to transfer the entire ConnectNetworkResponse (no fragmentation)? Maybe this should rather be triggered somehow from inside `BLEEndpoint` code that checks if `(mBtpEngine.TxState() == BtpEngine::kState_Complete)`.

Reply:
Updated to use BTP.